osf/migrations/0171_migrate_draft_registration_perms.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.15 on 2019-07-14 19:04
+from __future__ import unicode_literals
+
+import logging
+from django.db import migrations
+from django.core.management.sql import emit_post_migrate_signal
+from osf.models import DraftRegistration
+
+logger = logging.getLogger(__name__)
+
+
+def post_migrate_signal(state, schema):
+    # this is to make sure that the draft registration permissions created earlier exist!
+    emit_post_migrate_signal(3, False, 'default')
+
+
+add_draft_read_write_admin_auth_groups = """
+    INSERT INTO auth_group (name)
+    SELECT regexp_split_to_table('draft_registration_' || D.id || '_read,draft_registration_' || D.id || '_write,draft_registration_' || D.id || '_admin', ',')
+    FROM osf_draftregistration D;
+    """
+
+remove_draft_auth_groups = """
+    DELETE FROM auth_group WHERE name in
+    (SELECT regexp_split_to_table('draft_registration_' || D.id || '_read,draft_registration_' || D.id || '_write,draft_registration_' || D.id || '_admin', ',')
+    FROM osf_draftregistration D);
+    """
```

Comment:
Before auth_groups can be deleted, you need to also make sure all other constraints are taken care of.  DraftRegistrationGroupObjectPermissions table needs to be cleared (because that has pointers to this table), plus osf_osfusergroups rows that has any pointers to auth_group needs to be removed.

Haven't gotten to this yet - see guardian reverse migration (which did this exact same thing)